app/models/concerns/turbo/broadcastable.rb

```diff
@@ -296,8 +296,13 @@ def broadcast_rendering_with_defaults(options)
       options.tap do |o|
         # Add the current instance into the locals with the element name (which is the un-namespaced name)
         # as the key. This parallels how the ActionView::ObjectRenderer would create a local variable.
-        o[:locals] = (o[:locals] || {}).reverse_merge!(model_name.element.to_sym => self)
-        o[:partial] ||= to_partial_path
+        if o[:html].present? 
```

Comment:
I'd imagine that passing `html:` as an option is an edge case already, but it's worth highlighting that `html:` expects [valid and already escaped HTML text](https://guides.rubyonrails.org/layouts_and_rendering.html#rendering-html):

> You can send an HTML string back to the browser by using the :html option to render:
>
>     render html: helpers.tag.strong('Not Found')
>
> This is useful when you're rendering a small snippet of HTML code. However, you might want to consider moving it to a template file if the markup is complex.
> 
> When using html: option, HTML entities will be escaped if the string is not composed with html_safe-aware APIs.

If you want to use templating or mix helper calls with hard coded HTML `String` instances, the [`inline:` option](https://guides.rubyonrails.org/layouts_and_rendering.html#using-render-with-inline supports ERB:

> The render method can do without a view completely, if you're willing to use the :inline option to supply ERB as part of the method call. This is perfectly valid:
>
>    render inline: "<% products.each do |p| %><p><%= p.name %></p><% end %>"
>
> There is seldom any good reason to use this option. Mixing ERB into your controllers defeats the MVC orientation of Rails and will make it harder for other developers to follow the logic of your project. Use a separate erb view instead.

With that being said, would this commit's tests pass _without_ this implementation change? `o[:html] = o[:html]` is a no-operation, and the code without these changes already passes the `o` Hash through to the underlying `render` call.

Am I misunderstanding something else at play here?

Reply:
Yes this is because by default if you didn't pass in a partial it will use #to_partial_path on the object with these changes you would be able to pass in html: "<p> </p>" and this would get rendered currently it will throw an error couldn't find partial. This is going to be useful for use with view_component as well because you can just render the view_component and stick it as the html argument

Reply:
I understand now, the crucial part of the proposed change is the conditional, and not the contents of the block.

In that case, what do you think about guarding the `locals` and `partial` assignment with something more explicit, and omitting the `o[:html] = o[:html]`? Maybe something like:

```ruby
unless o.include?(:html) || o.include?(:inline)
  o[:locals] = (o[:locals] || {}).reverse_merge!(model_name.element.to_sym => self)
  o[:partial] ||= to_partial_path
end
```

Reply:
Right I see what you mean that would work well as it just passes the hash along

Reply:
also with the added bonus of supporting inline right away